MazerPlatformer/Character.cs

```diff
@@ -143,25 +150,29 @@ private void SetAnimationDirection(CharacterDirection direction)
                     Animation.CurrentAnimationDirection = Animation.AnimationDirection.Left;
                     break;
                 default:
-                    throw new ArgumentOutOfRangeException(nameof(direction), direction, null);
+                    return new InvalidDirectionFailure(direction);
             }
+
+            return Nothing;
         }
 
         // I can do unique things when my state changes
-        private void OnMyStateChanged(CharacterStates state)
+        private Either<IFailure, Unit> OnMyStateChanged(CharacterStates state)
         {
             if (state == CharacterStates.Idle) Animation.Idle = true;
+            return Nothing;
```

Comment:
Insteading of returning nothing or a Unit, which is a stop-gap from converting a void returning into an either, we can pass in the Animation, set it and pass it back without changing the accessing the it as a member on the class.